File: src/main/java/com/zuminX/names/SwaggerAnnotation.java
package com.zuminX.names;

/**
 * Swagger注解类名
 */
public class SwaggerAnnotation extends ClassName {

  public static final SwaggerAnnotation API;
  public static final SwaggerAnnotation API_IMPLICIT_PARAM;
  public static final SwaggerAnnotation API_IMPLICIT_PARAMS;
  public static final SwaggerAnnotation API_MODEL;
  public static final SwaggerAnnotation API_MODEL_PROPERTY;
  public static final SwaggerAnnotation API_OPERATION;
  public static final SwaggerAnnotation AUTHORIZATION;
  public static final SwaggerAnnotation AUTHORIZATION_SCOPE;
  public static final SwaggerAnnotation EXAMPLE;
  public static final SwaggerAnnotation EXAMPLE_PROPERTY;
  public static final SwaggerAnnotation EXTENSION;
  public static final SwaggerAnnotation EXTENSION_PROPERTY;
  public static final SwaggerAnnotation RESPONSE_HEADER;

  static {
    API = new SwaggerAnnotation("io.swagger.annotations.Api");
    API_IMPLICIT_PARAM = new SwaggerAnnotation("io.swagger.annotations.ApiImplicitParam");
    API_IMPLICIT_PARAMS = new SwaggerAnnotation("io.swagger.annotations.ApiImplicitParams");
    API_MODEL = new SwaggerAnnotation("io.swagger.annotations.ApiModel");
    API_MODEL_PROPERTY = new SwaggerAnnotation("io.swagger.annotations.ApiModelProperty");
    API_OPERATION = new SwaggerAnnotation("io.swagger.annotations.ApiOperation");
    AUTHORIZATION = new SwaggerAnnotation("io.swagger.annotations.Authorization");
    AUTHORIZATION_SCOPE = new SwaggerAnnotation("io.swagger.annotations.AuthorizationScope");
    EXAMPLE = new SwaggerAnnotation("io.swagger.annotations.Example");
    EXAMPLE_PROPERTY = new SwaggerAnnotation("io.swagger.annotations.ExampleProperty");
    EXTENSION = new SwaggerAnnotation("io.swagger.annotations.Extension");
    EXTENSION_PROPERTY = new SwaggerAnnotation("io.swagger.annotations.ExtensionProperty");
    RESPONSE_HEADER = new SwaggerAnnotation("io.swagger.annotations.ResponseHeader");
  }

  public SwaggerAnnotation(String qualifiedName) {
    super(qualifiedName);
  }

}

File: src/main/java/com/zuminX/names/annotation/SwaggerAnnotation.kt
package com.zuminX.names.annotation

import com.zuminX.names.ClassName

/**
 * Swagger注解类名
 */
class SwaggerAnnotation(qualifiedName: String) : ClassName(qualifiedName) {
  companion object {
    /**
     * 获取所有的Swagger注解类名列表
     *
     * @return Swagger注解类名列表
     */
    @JvmStatic
    fun getAll() = listOf(API, API_IMPLICIT_PARAM, API_IMPLICIT_PARAMS, API_MODEL, API_MODEL_PROPERTY, API_OPERATION,
      AUTHORIZATION, AUTHORIZATION_SCOPE, EXAMPLE, EXAMPLE_PROPERTY, EXTENSION, EXTENSION_PROPERTY, RESPONSE_HEADER)
  }
}

val API = SwaggerAnnotation("io.swagger.annotations.Api")
val API_IMPLICIT_PARAM = SwaggerAnnotation("io.swagger.annotations.ApiImplicitParam")
val API_IMPLICIT_PARAMS = SwaggerAnnotation("io.swagger.annotations.ApiImplicitParams")
val API_MODEL = SwaggerAnnotation("io.swagger.annotations.ApiModel")
val API_MODEL_PROPERTY = SwaggerAnnotation("io.swagger.annotations.ApiModelProperty")
val API_OPERATION = SwaggerAnnotation("io.swagger.annotations.ApiOperation")
val AUTHORIZATION = SwaggerAnnotation("io.swagger.annotations.Authorization")
val AUTHORIZATION_SCOPE = SwaggerAnnotation("io.swagger.annotations.AuthorizationScope")
val EXAMPLE = SwaggerAnnotation("io.swagger.annotations.Example")
val EXAMPLE_PROPERTY = SwaggerAnnotation("io.swagger.annotations.ExampleProperty")
val EXTENSION = SwaggerAnnotation("io.swagger.annotations.Extension")
val EXTENSION_PROPERTY = SwaggerAnnotation("io.swagger.annotations.ExtensionProperty")
val RESPONSE_HEADER = SwaggerAnnotation("io.swagger.annotations.ResponseHeader")
